niceml/data/datasets/dfdataset.py

```diff
@@ -27,51 +38,148 @@ class RegDataInfo(DataInfo):
+        self.dataframe_filters = dataframe_filters or []
+        self.df_path = df_path
+        self.data_location = data_location
         self.batch_size = batch_size
         self.set_name = set_name
         self.id_key = id_key
-        self.dataframe: pd.DataFrame = df_loader.load_df(df_path)
-        self.index_list = list(self.dataframe.index)
+        self.index_list = []
         self.shuffle = shuffle
+        self.data: Optional[pd.DataFrame] = None
         self.inputs: List[dict] = []
         self.targets: List[dict] = []
-
-    def get_batch_size(self) -> int:
-        return self.batch_size
-
-    def get_set_name(self) -> str:
-        return self.set_name
+        self.extra_key_list: List[str] = extra_key_list or []
+        self.feature_combiners: List[FeatureCombiner] = feature_combiners or []
 
     def initialize(
         self, data_description: RegDataDescription, exp_context: ExperimentContext
     ):
+        """
+        The initialize function is called when the data set is created.
+        It takes in a `RegDataDescription` object, which contains information about the
+        inputs and targets of your dataset. The initialize function should also take in an
+        `ExperimentContext` object, which contains information about where to find your
+        data on disk. The `ExperimentContext` is not used in this class.
+        This function should read in all the necessary files from disk and store them as
+        attributes on this class instance.
+
+        Args:
+            data_description: RegDataDescription: Pass the data description of the dataset
+                                to this class
+            exp_context: ExperimentContext: Pass the experiment context.
+        """
         self.inputs = data_description.inputs
         self.targets = data_description.targets
+
+        with open_location(self.data_location) as (data_fs, data_root):
+            data_path = join_fs_path(
+                data_fs, data_root, self.df_path.format(set_name=self.set_name)
+            )
+            self.data = read_parquet(filepath=data_path, file_system=data_fs)
+
+        for feature_combiner in self.feature_combiners:
+            self.data = feature_combiner.combine_features(self.data)
+
+        for df_filter in self.dataframe_filters:
+            df_filter.initialize(data_description=data_description)
+            self.data = df_filter.filter(data=self.data)
+
+        self.data = self.data.reset_index(drop=True)
+        self.index_list = list(range(len(self.data)))
+
         self.on_epoch_end()
 
+    def get_batch_size(self) -> int:
+        """
+        The get_batch_size function returns the batch size of the dataset.
+
+        Returns:
+            The batch size
+        """
+        return self.batch_size
+
+    def get_set_name(self) -> str:
+        """
+        The get_set_name function returns the name of the set.
+
+        Returns:
+            The `set_name` attribute
```

Comment:
Suggestion to stay consistent with the 'get_batch_size' function:
"The name of the data set"

Reply:
Done